### PR TITLE
fixed npy problem, refactored recog script

### PR DESCRIPTION
**Pull request by katebrune**
figured out how to get nparrays out of db without changing their data:
image

to figure this out, i refactored RecogScript.process_video_to_encode() by putting stuff in separate functions so i could individually test what was going on and pinpoint the problem. these functions are in FacialRecog.py (get_frame, get_face_encodings, compare_encodings). I also added tests for these functions in tests/facial_recog_tests.py.

additionally, i started looking over check_encodings in RecogScript and realized when we don't know who is in an image we would just add one person, but it could be two, so i made a function get_images_and_encodings (in FacialRecog.py) that takes a frame and returns encodings and images cropped around individual faces in the frame. This could be problematic if one person in the frame is in the system and one person isn't, because it would add both, so we should eventaully move this so we break the frame up by faces (cropped images) before we begin the facial recog comparisons.

### CODE REVIEW

File: RecogScript.py [hunk 1]
```diff
@@ -63,23 +41,27 @@ def check_encodings(all_encodings, all_ids, small_frame, temp_filename="images/t
...
-                    encoding = face_recognition.face_encodings(temp)
-                    encoding = str(encoding)
-                    data = {"img": encoded_image, "npy": encoding}
-                    add_unknown_person(data)
+
+                    frame = cv2.resize(image, (0, 0), fx=0.75, fy=0.75)
+                    unknown_encodings, unknown_images = get_images_and_encodings(frame)
+
+                    i = 0
+                    for encoding in unknown encodings:
```
Comment (gould-ann) [inline RecogScript.py]: heads up-- this is invalid syntax

**Review by gould-ann — CHANGES_REQUESTED** [review]

File: FacialRecog.py [hunk 2]
```diff
@@ -60,39 +60,80 @@ def scan_for_known_people_from_db(npy_known: str) -> dict:
...
+"""
 def get_all_people_information() -> Tuple[list, list]:
     all_encodings = []
     all_ids = []
 
     db_data = get_all_people("not_api_call")
     for i in db_data:
-        npy = np.fromstring(i['npy'], count=128)
+        formatted_npy_str = i['npy'].replace(r"\n", "")[2:-2] #get rid on newlines and "[]"
+        iterable = formatted_npy_str.split() #make string values iterable
```
Comment (gould-ann) [inline FacialRecog.py:73]: also good job on this!

**Review by gould-ann — APPROVED** [review]
😍 i didnt know i could love code until I saw this code

**Review by rgoluch — COMMENTED** [review]
awesome job! looks great!

**Pull request merged**